plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/KVMHAMonitor.java

```diff
@@ -35,121 +35,189 @@
 public class KVMHAMonitor extends KVMHABase implements Runnable {
 
     private static final Logger s_logger = Logger.getLogger(KVMHAMonitor.class);
-    private final Map<String, NfsStoragePool> storagePool = new ConcurrentHashMap<>();
+    private final Map<String, NfsStoragePool> nfsstoragePool = new ConcurrentHashMap<>();
+    private final Map<String, RbdStoragePool> rbdstoragePool = new ConcurrentHashMap<>();
+    private final NfsStoragePool nfsStoragePool = null;
+    private final RbdStoragePool rbdStoragePool = null;
     private final boolean rebootHostAndAlertManagementOnHeartbeatTimeout;
 
     private final String hostPrivateIp;
 
-    public KVMHAMonitor(NfsStoragePool pool, String host, String scriptPath) {
+    public KVMHAMonitor(NfsStoragePool pool, RbdStoragePool rbdpool, String host, String scriptPath, String scriptPathRbd) {
         if (pool != null) {
-            storagePool.put(pool._poolUUID, pool);
+            nfsstoragePool.put(pool._poolUUID, pool);
+        }else if (rbdpool != null) {
+            rbdstoragePool.put(rbdpool._poolUUID, rbdpool);
         }
+        configureHeartBeatPath(scriptPath, scriptPathRbd);
         hostPrivateIp = host;
-        configureHeartBeatPath(scriptPath);
-
         _heartBeatUpdateTimeout = AgentPropertiesFileHandler.getPropertyValue(AgentProperties.HEARTBEAT_UPDATE_TIMEOUT);
         rebootHostAndAlertManagementOnHeartbeatTimeout = AgentPropertiesFileHandler.getPropertyValue(AgentProperties.REBOOT_HOST_AND_ALERT_MANAGEMENT_ON_HEARTBEAT_TIMEOUT);
     }
 
-    private static synchronized void configureHeartBeatPath(String scriptPath) {
+    private static synchronized void configureHeartBeatPath(String scriptPath, String scriptPathRbd) {
         KVMHABase.s_heartBeatPath = scriptPath;
+        KVMHABase.s_heartBeatPathRbd = scriptPathRbd;
     }
 
     public void addStoragePool(NfsStoragePool pool) {
-        synchronized (storagePool) {
-            storagePool.put(pool._poolUUID, pool);
+        synchronized (nfsstoragePool) {
```

Comment:
Since you are already using `ConcurrentHashMap`, I dont see the need of synchronized